### PR TITLE
Added support for react-native mobile apps

### PR DESCRIPTION
**Pull request by vedyed**
Added react-native async-storage support for setting and getting items. The current storage provider uses `window.localStorage` which is applicable for websites and doesn't work for mobile apps. 

Tested and verified.

### CODE REVIEW

File: src/storage-provider-local.ts [hunk 1]
```diff
@@ -7,6 +8,14 @@ export default class LocalStorageProvider implements IStorageProvider {
...
             const key = `${this.prefix}:${name}`;
+            try {
+                AsyncStorage.setItem(key, repo).then((e: any) => {
+                    console.log(`Unleash local save completed for ${key} with value ${repo} and response is ${JSON.stringify(e)}`);
+                });
+            } catch(ex) {
+                console.error(ex);  
+            }
+            // below line is not needed for react-native mobile app
             window.localStorage.setItem(key, repo);
```
Comment (ivarconr) [inline src/storage-provider-local.ts:19]: Should we just remove the `winow.localStorage` call all-together? I guess AsyncStorage is used on web as well?

Comment (vedyed) [reply src/storage-provider-local.ts:19]: Retained this line so that the same package can be used for both web and mobile apps.

Comment (vedyed) [reply src/storage-provider-local.ts:19]: We can remove this if you're goal is to have separate packages for web and mobile

Comment (vedyed) [reply src/storage-provider-local.ts:19]: I believe Async-storage cannot be used for websites

Comment (vedyed) [reply src/storage-provider-local.ts:19]: Suggestion: 
Create a new repo `unleash-proxy-client-react-native` from current `unleash-proxy-client-js` and then we can merge this PR to the newly created repo(we can remove window.localStorage as it is exclusive for react-native mobile apps)

**Comment by ivarconr** [comment]
I think Async Storage can be used on web as well. Need to validate.

https://github.com/react-native-async-storage/async-storage/releases/tag/v1.9.0

**Pull request closed without merging**